Shared/messages/MessagesOverlay.cpp

```diff
@@ -37,153 +37,59 @@ namespace Dsa {
+  \inherits DynamicEntityLayer
+  \brief Manages a \l Esri::ArcGISRuntime::DynamicEntityLayer object
   for displaying message feed graphics.
 
   The overlay currently only supports messages containing a
   point geometry type.
  */
 
 /*!
-  \brief Constructor taking a \a geoView and an optional \a parent.
+  \brief Constructor taking a \a messageFeed, a \a messageType
+  and an optional \a parent.
  */
-
-/*!
-  \brief Constructor taking a \a geoView, a \a renderer, a \a messageType,
-  a \a surfacePlacement mode and an optional \a parent.
- */
-
-/*!
-  \brief Destructor.
- */
-MessagesOverlay::MessagesOverlay(GeoView* geoView, Renderer* renderer, const QString& messageType, MessageFeed* messageFeed, SurfacePlacement surfacePlacement, QObject* parent) :
-  QObject(parent),
-  m_geoView(geoView),
-  m_renderer(renderer),
-  m_surfacePlacement(surfacePlacement),
-  m_messageFeed(messageFeed)
+MessagesOverlay::MessagesOverlay(MessageFeed* messageFeed, const QString& messageType, QObject* parent) :
+    DynamicEntityLayer(messageFeed, parent),
+    m_messageFeed(messageFeed)
 {
-  m_dynamicEntityLayer = new DynamicEntityLayer(m_messageFeed, this);
-  m_dynamicEntityLayer->setLayerId(messageType);
-  m_dynamicEntityLayer->setSceneProperties(LayerSceneProperties(m_surfacePlacement));
-  m_dynamicEntityLayer->setRenderer(m_renderer);
-  m_dynamicEntityLayer->setName(messageType);
-  SceneView* scene = static_cast<SceneView*>(m_geoView);
-  scene->arcGISScene()->operationalLayers()->append(m_dynamicEntityLayer);
-}
-
-MessagesOverlay::~MessagesOverlay()
-{
-}
-
-/*!
-  \brief Returns the Esri:ArcGISRuntime::Renderer in use by the overlay.
- */
-Renderer* MessagesOverlay::renderer() const
-{
-  return m_renderer.data();
-}
-
-/*!
-  \brief Sets the Esri:ArcGISRuntime::Renderer for the overlay to \a renderer.
- */
-void MessagesOverlay::setRenderer(Renderer* renderer)
-{
-  if (m_renderer == renderer)
-    return;
-
-  m_renderer = renderer;
-
-  m_dynamicEntityLayer->setRenderer(m_renderer);
+  this->setLayerId(messageType);
+  this->setName(messageType);
+  messageFeed->setMessagesOverlay(this);
 }
 
 /*!
-  \brief Returns the Esri:ArcGISRuntime::SurfacePlacement for the overlay.
- */
-SurfacePlacement MessagesOverlay::surfacePlacement() const
-{
-  return m_surfacePlacement;
-}
-
-/*!
-  \brief Sets the Esri:ArcGISRuntime::SurfacePlacement for the overlay to \a surfacePlacement.
+  \brief Destructor.
  */
-void MessagesOverlay::setSurfacePlacement(SurfacePlacement surfacePlacement)
-{
-  if (m_surfacePlacement == surfacePlacement)
-    return;
-
-  m_surfacePlacement = surfacePlacement;
-
-  m_dynamicEntityLayer->setSceneProperties(LayerSceneProperties(m_surfacePlacement));
-}
+MessagesOverlay::~MessagesOverlay() = default;
 
 /*!
   \brief Returns the message type for the overlay.
  */
 QString MessagesOverlay::messageType() const
 {
-  return m_dynamicEntityLayer->layerId();
+  return this->layerId();
```

Comment:
is `this->` needed? Same question below.

Reply:
removed.